packages/go_router/test/custom_transition_page_test.dart

```diff
@@ -117,6 +117,45 @@ void main() {
     await tester.pumpAndSettle();
     expect(find.byKey(homeKey), findsOneWidget);
   });
+
+  testWidgets('transitionDuration and reverseTransitionDuration is different',
+      (WidgetTester tester) async {
+    const ValueKey<String> homeKey = ValueKey<String>('home');
+    const ValueKey<String> loginKey = ValueKey<String>('login');
+    const Duration transitionDuration = Duration(milliseconds: 50);
+    const Duration reverseTransitionDuration = Duration(milliseconds: 500);
+
+    final GoRouter router = GoRouter(
+      routes: <GoRoute>[
+        GoRoute(
+          path: '/',
+          builder: (_, __) => const HomeScreen(key: homeKey),
+        ),
+        GoRoute(
+          path: '/login',
+          pageBuilder: (_, GoRouterState state) => CustomTransitionPage<void>(
+            key: state.pageKey,
+            transitionDuration: transitionDuration,
+            reverseTransitionDuration: reverseTransitionDuration,
+            transitionsBuilder:
+                (_, Animation<double> animation, ___, Widget child) =>
+                    FadeTransition(opacity: animation, child: child),
+            child: const LoginScreen(key: loginKey),
+          ),
+        ),
+      ],
+    );
+    await tester.pumpWidget(MaterialApp.router(routerConfig: router));
+    expect(find.byKey(homeKey), findsOneWidget);
+
+    router.push('/login');
+    await tester.pumpAndSettle();
+    expect(find.byKey(loginKey), findsOneWidget);
+
+    router.pop();
+    await tester.pumpAndSettle();
+    expect(find.byKey(homeKey), findsOneWidget);
+  });
```

Comment:
Having trouble coming up with a test that asserts `transitionDuration` and `reverseTransitionDuration` is different... Any suggestions?

Reply:
Just throwing out idea. In you transition builder, can you build a text widget with animation value, and in the test you can use tester.pump(Duration(milliseconds: 100)) and make sure the animation value is close to reverseDuration/100ms

Reply:
Thanks for the suggestions! I ended up finding a simpler approach:
``` dart
router.push('/login');
final int pushingPumped = await tester.pumpAndSettle();
expect(find.byKey(loginKey), findsOneWidget);

router.pop();
final int poppingPumped = await tester.pumpAndSettle();
expect(find.byKey(homeKey), findsOneWidget);

expect(pushingPumped != poppingPumped, true);
```